cli/src/lib.rs

```diff
@@ -187,35 +183,7 @@ where
 		ParseAndPrepare::RevertChain(cmd) => {
 			cmd.run_with_builder(|config: Config<_, _>| Ok(new_full_start!(config).0), load_spec)
 		}
-		ParseAndPrepare::CustomCommand(CustomSubcommands::Factory(cli_args)) => {
-			let mut config: Config<_, _> =
-				cli::create_config_with_db_path(load_spec, &cli_args.shared_params, &version)?;
-			config.execution_strategies = ExecutionStrategies {
-				importing: cli_args.execution.into(),
-				block_construction: cli_args.execution.into(),
-				other: cli_args.execution.into(),
-				..Default::default()
-			};
-
-			match ChainSpec::from(config.chain_spec.id()) {
-				Some(ref c) if c == &ChainSpec::Development || c == &ChainSpec::LocalTestnet => {}
-				_ => panic!("Factory is only supported for development and local testnet."),
-			}
-
-			let factory_state = FactoryState::new(cli_args.mode.clone(), cli_args.num, cli_args.rounds);
-
-			let service_builder = new_full_start!(config).0;
-			transaction_factory::factory::<FactoryState<_>, _, _, _, _, _>(
-				factory_state,
-				service_builder.client(),
-				service_builder
-					.select_chain()
-					.expect("The select_chain is always initialized by new_full_start!; QED"),
-			)
-			.map_err(|e| format!("Error in transaction factory: {}", e))?;
-
-			Ok(())
-		}
+		ParseAndPrepare::CustomCommand(_) => Ok(()),
```

Comment:
I removed it because there was some compile errors and I thought we didn't need it. But the error is fixed and it's re-instated along with its dependency `factory_impl.rs`